packages/reakit/src/Dialog/__examples__/DialogWithForm/__tests__/index-test.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+import { render, fireEvent, type, wait, click } from "reakit-test-utils";
+import DialogWithForm from "..";
+
+test("should rename", async () => {
+  const { getByText, getByLabelText, getByRole } = render(<DialogWithForm />);
+  const name = getByText("Name");
+  expect(name).toBeVisible();
+  const disclosure = getByText("✏️");
+  const dialog = getByLabelText("Choose a new name");
+  expect(dialog).not.toBeVisible();
+  click(disclosure);
+  expect(dialog).toBeVisible();
+  const form = getByRole("form");
+  const input = getByLabelText("New name");
+  type(" 2", input);
+  fireEvent.submit(form);
```

Comment:
I would try this (need to import `focus` and `press` from `reakit-test-utils`):
```suggestion
  focus(input);
  type(" 2");
  press.Enter();
```